R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
+getPKResultsDataFrame <- function(pkParameterResultsFilePath) {
+  pkResultsDataFrame <- read.csv(pkParameterResultsFilePath, encoding = "UTF-8", check.names = FALSE)
+  colnames(pkResultsDataFrame) <- c("IndividualId", "QuantityPath", "Parameter", "Value", "Unit")
+  pkResultsDataFrame$QuantityPath <- as.factor(pkResultsDataFrame$QuantityPath)
+  pkResultsDataFrame$Parameter <- as.factor(pkResultsDataFrame$Parameter)
+  pkResultsDataFrame$Unit <- as.factor(pkResultsDataFrame$Unit)
+  return(pkResultsDataFrame)
+}
+
+
+
+
+#' @title getQuantileIndividualIds
+#' @description Find IDs of individuals whose PK analysis results closest to quantiles given by vector of quantiles quantileVec
+#' @param pkAnalysisResultsDataframe Dataframe storing the PK analysis results for multiple individuals for a single PK parameter and single output path
+#' @return ids, IDs of individuals whose PK analysis results closest to quantiles given by vector of quantiles quantileVec
+getQuantileIndividualIds <- function(pkAnalysisResultsDataframe, quantileVec = c(0.05, 0.5, 0.95)) {
+  rowNums <- NULL
+  for (n in 1:length(quantileVec)) {
+    rowNums[n] <- which.min(abs(pkAnalysisResultsDataframe$Value - quantile(pkAnalysisResultsDataframe$Value, quantileVec[n])))
+  }
+  ids <- as.numeric(pkAnalysisResultsDataframe$IndividualId[rowNums])
+  values <- pkAnalysisResultsDataframe$Value[rowNums]
+  units <- as.character(pkAnalysisResultsDataframe$Unit[rowNums])
+  quantileResults <- list(ids = ids,values=values,units=units)
+  return(quantileResults)
+}
+
+#' @title populationSensitivityAnalysis
+#' @param simFilePath path to simulation file
+#' @param popDataFilePath path to population file
+#' @param pkParameterResultsFilePath path to pk parameter results CSV
+#' @param resultsFileFolder folder where results of sensitivity analysis will be saved
+#' @param popSAResultsIndexFile Names of CSV file containing index of sensitivity analysis CSV files
+#' @param resultsFileName root name of sensitivity analysis results CSV files
+#' @param quantileVec vector of quantiles in (0,1).  For each output and pk parameter, there will be a distribution of pk parameter values for the population.
+#' The individuals yielding pk parameters closest to these quantiles will be selected for sensitivity analysis.  Default is (0.05,0.5,0.95)
+#' @param numberOfCores the number of cores to be used for parallelization of the sensitivity analysis.  Default is 1 core (no parallelization).
+#' @export
+populationSensitivityAnalysis <- function(simFilePath,
+                                          popDataFilePath,
+                                          pkParameterResultsFilePath,
+                                          resultsFileFolder,
+                                          resultsFileName,
+                                          popSAResultsIndexFile = "sensitivityAnalysesResultsIndexFile",
+                                          quantileVec = c(0.05, 0.5, 0.95),
```

Comment:
We duplicate the same default values here. This should be in only one method and the other one should require inputs in that case

Reply:
default is now only specified at the workflow level, rather than in the function